quinn/src/platform/unix.rs

```diff
@@ -198,9 +198,25 @@ fn send(io: &mio::net::UdpSocket, transmits: &[Transmit]) -> io::Result<usize> {
     let mut msgs: [libc::mmsghdr; BATCH_SIZE] = unsafe { mem::zeroed() };
     let mut iovecs: [libc::iovec; BATCH_SIZE] = unsafe { mem::zeroed() };
     let mut cmsgs = [cmsg::Aligned([0u8; CMSG_LEN]); BATCH_SIZE];
+    // This assume_init looks a bit weird because one might think it
+    // assumes the SockAddr data to be initialized, but that call
+    // refers to the whole array, which itself is made up of MaybeUninit
+    // containers. Their presence protects the SockAddr inside from
+    // being assumed as initialized by the assume_init call.
+    // TODO: Replace this with uninit_array once it becomes MSRV-stable
+    let mut addrs: [MaybeUninit<socket2::SockAddr>; BATCH_SIZE] =
+        unsafe { MaybeUninit::uninit().assume_init() };
     for (i, transmit) in transmits.iter().enumerate().take(BATCH_SIZE) {
+        let dst_addr = unsafe {
+            std::ptr::write(
+                addrs[i].as_mut_ptr(),
+                socket2::SockAddr::from(transmit.destination),
+            );
+            &mut *(&mut addrs[i] as *mut _ as *mut _)
```

Comment:
Looks like we only pass this as a shared reference. This can also be a lot more concise.
```suggestion
            &*addrs[i].as_ptr()
```

Reply:
Good catch. The old code was a relic from the times when I still passed a mutable reference.